src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -121,12 +131,63 @@ function useChatUtilsHook(): UseChatUtilsHook {
     });
   };
 
+  const helperSaveChannelDetail = async (channel: ChannelList, response: any) => {
+    if (!localDb) return;
+    const {originalMembers} = getChannelListInfo(response?.channel);
+
+    const promise = originalMembers?.map((member) => {
+      return new Promise((resolve, reject) => {
+        const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+        try {
+          user.saveOrUpdateIfExists(localDb).then(() => resolve(null));
+        } catch (e) {
+          reject(e);
+        }
+      });
+    });
+
+    await Promise.all(promise).catch((e) => console.log('saveChannelDetail error', e));
+  };
+
+  const helperGetChannelDetail = async (channel: ChannelList) => {
+    if (!localDb) return;
+
+    console.log('should fetch channel detail');
+    try {
+      let response;
+      if (channel?.channelType === 'ANON_PM') {
+        response = await AnonymousMessageRepo.getAnonymousChannelDetail('messaging', channel?.id);
+      } else {
+        const channelType = channel?.channelType === 'GROUP' ? 'group' : 'messaging';
+        response = await SignedMessageRepo.getSignedChannelDetail(channelType, channel?.id);
+      }
+      await helperSaveChannelDetail(channel, response);
+    } catch (e) {
+      console.log('getChannelDetail error', e);
+    } finally {
+      setChat((prevChat) => ({
+        ...prevChat,
+        isLoadingFetchingChannelDetail: false
+      }));
+      refresh('channelInfo');
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new function `helperGetChannelDetail` is introduced to fetch channel details based on the channel type and ID. This function also handles potential errors and sets the loading state accordingly, which is a good practice. However, similar to the previous comment, consider handling errors in a more user-friendly way.
```diff
+ catch (e) {
+   // Instead of just logging the error, consider showing an error message to the user or retrying the operation
+   console.log('getChannelDetail error', e);
+ }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->